pkg/core/resources/apis/mesh/dataplane_overview_test.go

```diff
@@ -96,4 +96,62 @@ var _ = Describe("DataplaneOverview", func() {
 				expected: DataplaneOverviewResourceList{Items: []*DataplaneOverviewResource{}},
 			}))
 	})
+	Describe("RetainGateWayDataPlanes", func() {
+		gateWayDataPlanes := DataplaneOverviewResourceList{
+			Items: []*DataplaneOverviewResource{
+				{
+					Spec: v1alpha1.DataplaneOverview{
+						Dataplane: &v1alpha1.Dataplane{
+							Networking: &v1alpha1.Dataplane_Networking{
+								Gateway: &v1alpha1.Dataplane_Networking_Gateway{
+									Tags: map[string]string{
+										"service": "gateway",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		dataPlanes := DataplaneOverviewResourceList{
+			Items: []*DataplaneOverviewResource{
+				{
+					Spec: v1alpha1.DataplaneOverview{
+						Dataplane: &v1alpha1.Dataplane{
+							Networking: &v1alpha1.Dataplane_Networking{
+								Inbound: []*v1alpha1.Dataplane_Networking_Inbound{
+									{
+										Tags: map[string]string{
+											"service": "mobile",
+											"version": "v1",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		type testCase struct {
+			actual   DataplaneOverviewResourceList
+			expected DataplaneOverviewResourceList
+		}
+		DescribeTable("should retain gateway overviews", func(given testCase) {
+			// when
+			given.actual.RetainGatewayDataPlanes()
+
+			// then
+			Expect(given.actual).To(Equal(given.expected))
+		},
+			Entry("should retain all gateway dataplanes", testCase{
+				actual:   gateWayDataPlanes,
+				expected: gateWayDataPlanes,
+			}),
+			Entry("should retain none if no gateway dataplanes present", testCase{
+				actual:   dataPlanes,
+				expected: DataplaneOverviewResourceList{Items: []*DataplaneOverviewResource{}},
+			}))
```

Comment:
How about instead of having two test cases have one test with a list that has 1 regular DP and 1 gateway DP and see if only one is retained?

Reply:
can I add this in additional to the above 2 test cases?

Reply:
sure although I can't see much value of having 3 tests over 1 if the 1 actually checks the same thing what 3 of them would do.

Reply:
agreed on that. added only the test case with 1 regular DP and 1 Gateway DP

Reply:
If you've got only one case, I think it's better to fallback this to the regular `It()` test without the table